File: game_server/GameRes/src/NPGameRes/GameObjs/PlayerCondition/ConditionObj/NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM.java
package NPGameRes.GameObjs.PlayerCondition.ConditionObj;

import ALServerLog.ALServerLog;
import NPCommon.CommonObj.NPStringReader;
import NPEnum.ENPPlayerConditionType;
import NPGameRes.GameObjs.PlayerCondition._ANPBasicPlayerCondition;

/**
 * 火星空闲人口数量判断:min(:max) CS_MARS_IDLE_PEOPLE_NUM
 * @author claude
 */
public class NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM extends _ANPBasicPlayerCondition
{
    private long _m_lMinValue; // -1:不限制最小值
    private long _m_lMaxValue = -1; //-1:不限制最大值

    public long minValue() {return _m_lMinValue;}
    public long maxValue() {return _m_lMaxValue;}

    @Override
    public ENPPlayerConditionType conditionType()
    {
        return ENPPlayerConditionType.CS_MARS_IDLE_PEOPLE_NUM;
    }

    public static NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM readStr(NPStringReader _reader)
    {
        NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM cond = new NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM();

        String minVS = _reader.readItem();

        if (null == minVS)
        {
            ALServerLog.Error("Can not read str for ENPPlayerConditionType.CS_MARS_IDLE_PEOPLE_NUM[" + _reader.getSrcString() + "]");
            return null;
        }

        cond._m_lMinValue = Long.parseLong(minVS);

        String maxVS = _reader.readItem();
        if (null != maxVS)
        {
        	cond._m_lMaxValue = Long.parseLong(maxVS);
        }

        return cond;
    }
}

File: game_client/Assets/Scripts/ResScripts/NPResCommon/PlayerCondition/ConditionObj/NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM.cs
using NPEnum;
using ALPackage;

namespace GOE
{
    /// <summary>
    /// 火星空闲人口数量判断 CS_MARS_IDLE_PEOPLE_NUM:min_value（:max_value）
    /// </summary>
    public class NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM : _ANPBasicPlayerCondition
    {
        public override ENPPlayerConditionType conditionType { get { return ENPPlayerConditionType.CS_MARS_IDLE_PEOPLE_NUM; } }

        private WCGLongRange _m_lRange;

        public static NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM readStr(ALStringReader _reader)
        {
            NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM cond = new NPPlayerCondition_CS_MARS_IDLE_PEOPLE_NUM();

            string minVS = _reader.readItem(':');
            if (null != minVS)
            {
                string maxVS = _reader.readItem(':');
                if (null != maxVS)
                    cond._m_lRange = new WCGLongRange(minVS, maxVS);
                else
                    cond._m_lRange = new WCGLongRange(minVS, "-1");
            }
            else
            {
                cond._m_lRange = new WCGLongRange(1, -1);
            }

            return cond;
        }

        public override bool isEnable(NPVarInfo _varVariableInfo)
        {
#if NP_GAME
            return isRange(NPPlayer.instance.marsComp.peopleSubComponent.idlePeopleNum, _m_lRange.min, _m_lRange.max);
#else
            return false;
#endif
        }
    }
}